### PR TITLE
uncomitted heavy files, added profanity filter

### PR DESCRIPTION
**Pull request by ady00**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

Added a simple profanity filter through the bad-words lib. Extension to #3238.

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: src/components/ProblemSolutions.tsx [hunk 1]
```diff
@@ -203,7 +205,9 @@ export default function ProblemSolutions({
                             : undefined
                         }
                       >
-                        {submission.solutionCode}
+                        {filter.isProfane(submission.solutionCode)
+                          ? filter.clean(submission.solutionCode)
```
Comment (ady00) [inline src/components/ProblemSolutions.tsx:209]: Tried filter.clean and filter.isProfane on my personal build, filter.isProfane was significantly faster on sections with more than 30-40 public user solutions.

Comment (SansPapyrus683) [reply src/components/ProblemSolutions.tsx:209]: using `filter.clean` on everything makes performance take a hit right?

Comment (ady00) [reply src/components/ProblemSolutions.tsx:209]: I use `filter.isProfane` on everything, and `filter.clean` is only reserved for submissions that triggered `filter.isProfane`.
`Filter.isProfane` is run on every submission, and I don't think there's a way around that.

Comment (thecodingwizard) [reply src/components/ProblemSolutions.tsx:209]: you can plausibly use React.useMemo() to avoid rerunning this expensive computation

Comment (ady00) [reply src/components/ProblemSolutions.tsx:209]: Isn't React.useMemo() only beneficial when the page rerenders numerous times? AFAIK this page only renders once when it loads in all the solutions, so using useMemo wouldn't optimize that initial render.

Comment (thecodingwizard) [reply src/components/ProblemSolutions.tsx:209]: if parent components rerender this component (might?) also rerender. you can probably test this just by logging to console during the render... (I could be wrong -- maybe it really only runs once?)

Comment (ady00) [reply src/components/ProblemSolutions.tsx:209]: Yeah, the submissions don't rerender whenever the user upvotes a submission or adds a submission (which are the only components which can change states on the problem-solutions page), meaning that submissions only render once

Comment (thecodingwizard) [reply src/components/ProblemSolutions.tsx:209]: UserDataContext can rerender (many) times I think

**Pull request merged**